tensorboard/plugins/projector/vz_projector/knn.ts

```diff
@@ -59,47 +59,41 @@ export function findKNNGPUCosine<T>(
   // A*A.T will give us NxN matrix holding the cosine distance between every
   // pair of points, which we sort using KMin data structure to obtain the
   // K nearest neighbors for each point.
-  let typedArray = vector.toTypedArray(dataPoints, accessor);
-  const bigMatrix = tf.tensor(typedArray, [N, dim]);
-  let nearest: NearestEntry[][] = new Array(N);
-  let numPieces = Math.ceil(N / OPTIMAL_GPU_BLOCK_SIZE);
+  const nearest: NearestEntry[][] = new Array(N);
+  const numPieces = Math.ceil(N / OPTIMAL_GPU_BLOCK_SIZE);
   let M = Math.floor(N / numPieces);
   let modulo = N % numPieces;
   let offset = 0;
   let progress = 0;
   let progressDiff = 1 / (2 * numPieces);
   let piece = 0;
+
+  const typedArray = vector.toTypedArray(dataPoints, accessor);
+  const bigMatrix = tf.tensor(typedArray, [N, dim]);
+  const bigMatrixTransposed = bigMatrix.transpose();
+  const bigMatrixSquared = tf.matMul(bigMatrix, bigMatrixTransposed);
+  const cosDist = tf.sub(1, bigMatrixSquared);
```

Comment:
nit: I realize that the `cosDist` here and the `cosDist` on L96 do not collide, but could we rename them since they represent different things?  Maybe this one on L75 could be `cosDistMatrix`?